autokeras/task.py

```diff
@@ -167,3 +167,176 @@ def __init__(self,
+                The model will set apart this fraction of the training data,
+                will not train on it, and will evaluate
+                the loss and any model metrics
+                on this data at the end of each epoch.
+                The validation data is selected from the last samples
+                in the `x` and `y` data provided, before shuffling. This argument is
+                not supported when `x` is a dataset.
+            validation_data: Data on which to evaluate
+                the loss and any model metrics at the end of each epoch.
+                The model will not be trained on this data.
+                `validation_data` will override `validation_split`.
+                `validation_data` could be:
+                  - tuple `(x_val, y_val)` of Numpy arrays or tensors
+                  - tuple `(x_val, y_val, val_sample_weights)` of Numpy arrays
+                  - dataset or a dataset iterator
+                For the first two cases, `batch_size` must be provided.
+                For the last case, `validation_steps` must be provided.
+                The type of the validation data should be the same as the training
+                data.
+            **kwargs: Any arguments supported by keras.Model.fit.
+        """
+        # x is file path of training data
+        if isinstance(x, str):
+            df = pd.read_csv(x)
+            validation_df = pd.read_csv(validation_data)
+            label = df.pop(y).to_numpy()
+            validation_label = validation_df.pop(y).to_numpy()
+            validation_data = (validation_df, validation_label)
+            x = df
+            y = label
+            validation_split = 0
+
+        super().fit(x=x,
+                    y=y,
+                    validation_split=validation_split,
+                    validation_data=validation_data,
+                    **kwargs)
+
+
+class StructuredDataClassifier(SupervisedStructuredDataPipeline):
+    """AutoKeras structured data classification class.
+
+    # Arguments
+        column_names: A list of strings specifying the names of the columns. The
+            length of the list should be equal to the number of columns of the data.
+            Defaults to None. If None, it will obtained from the header of the csv
+            file or the pandas.DataFrame.
+        column_types: Dict. The keys are the column names. The values should either
+            be 'numerical' or 'categorical', indicating the type of that column.
+            Defaults to None. If not None, the column_names need to be specified.
+            If None, it will be inferred from the data.
+        num_classes: Int. Defaults to None. If None, it will infer from the data.
+        multi_label: Boolean. Defaults to False.
+        loss: A Keras loss function. Defaults to None. If None, the loss will be
+            inferred from the AutoModel.
+        metrics: A list of Keras metrics. Defaults to None. If None, the metrics will
```

Comment:
What does it imply for the metrics to be inferred? Is the list of default metrics known in advance? (e.g. always Accuracy?)

Reply:
It would always be Accuracy for classification and always MSE for regression.
Yeah, the docstrings are confusing. I will change it.